### PR TITLE
Fix/multitest parting revision

### PR DESCRIPTION
**Pull request by zhenyu-ms**
## Bug / Requirement Description
~Clearly and concisely describe the problem.~

## Solution description
use testcase-level round-robin scheduling for multitest parting (maybe more to go)

## Checklist:
- [ ] Test
- [ ] Example (both test_plan.py and .rst)
- [ ] Documentation (API)
- [ ] News fragment present for release notes
- [x] MS info leakage check
- [ ] For new driver: driver index page
- [ ] For new assertion: ui/pdf/std renderers, documentation
- [ ] For new cmdline arg: documentation

### CODE REVIEW

File: testplan/testing/multitest/base.py [hunk 1]
```diff
@@ -425,26 +423,39 @@ def get_test_context(self):
...
-                        )
-                        data = self.cfg.xfail_tests.get(
-                            testcase_instance, None
-                        )
-                        if data is not None:
-                            testcase.__func__.__xfail__ = {
-                                "reason": data["reason"],
-                                "strict": data["strict"],
-                            }
+                for testcase in testcases_to_run:
```
Comment (Pyifan) [inline testplan/testing/multitest/base.py:426]: why are we making this change, i think the original code is fine.

**Review by Pyifan — CHANGES_REQUESTED** [review]
I think if we change the splitting logic, we need to change merge logic too?

File: testplan/testing/multitest/base.py (hunk 1 above)
Comment (zhenyu-ms) [reply testplan/testing/multitest/base.py:426]: ah i just want to remove extra if statements

File: testplan/testing/multitest/base.py [hunk 2]
```diff
@@ -392,6 +392,11 @@ def get_test_context(self):
         ctx = []
         sorted_suites = self.cfg.test_sorter.sorted_testsuites(self.cfg.suites)
 
+        if hasattr(self.cfg, "xfail_tests") and self.cfg.xfail_tests:
```
Comment (yuxuan-ms) [inline testplan/testing/multitest/base.py:395]: set self.cfg.xfail_tests default value to `{}` ?

File: testplan/common/report/base.py [hunk 3]
```diff
@@ -634,6 +640,27 @@ def remove_by_uid(self, uid):
...
+        yield self
+        for e in self:
+            if isinstance(e, BaseReportGroup):
+                yield from e.pre_order_reports()
+            elif isinstance(e, Report):
+                yield e
+
+    def disassemble(self):
+        c = []
+        for u in copy.copy(list(self._index.keys())):
```
[the remark this answers is not in the document]
Comment (zhenyu-ms) [reply testplan/common/report/base.py:653]: will fix 😵

File: testplan/testing/multitest/base.py (hunk 2 above)
Comment (zhenyu-ms) [reply testplan/testing/multitest/base.py:395]: (i'd like to pretend that) self.cfg is immutable 😄

**Review by Pyifan — APPROVED** [review]

**Pull request merged**